### ISSUE TITLE
FileNotFoundException - Too many open files

### ISSUE THREAD

**Issue by mhernand40** [amplitude/Amplitude-Kotlin#71]
Seeing the following crash in production for our Android app:
```
java.io.FileNotFoundException: /data/user/0/*********/app_amplitude-disk-queue/cab21925f6ed6f9bb0f53ea20cba9c2b-4.tmp (Too many open files)
    at java.io.FileOutputStream.open0(FileOutputStream.java)
    at java.io.FileOutputStream.open(FileOutputStream.java:287)
    at java.io.FileOutputStream.<init>(FileOutputStream.java:223)
    at com.amplitude.core.utilities.EventsFileManager.writeToFile(SourceFile:147)
    at com.amplitude.core.utilities.EventsFileManager.finish(SourceFile:128)
    at com.amplitude.core.utilities.EventsFileManager.rollover(SourceFile:99)
    at com.amplitude.android.utilities.AndroidStorage.rollover
    at com.amplitude.core.platform.EventPipeline$upload$1$1$1.invokeSuspend(SourceFile:102)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:33)
    at kotlinx.coroutines.DispatchedTask.run(SourceFile:106)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at java.lang.Thread.run(Thread.java:764)
```

Though the number of affected users seems to be low; less than 50 users or so.

## Environment
- Device OS and Version: Android 7.0, 7.1.1, 7.1.2, 8.0.0, 8.1.0, 9, 10, 11, 12

**Comment by yuhao900914** [comment]
Hi @mhernand40 , which version of SDK are you using?

**Comment by mhernand40** [comment]
Hi @yuhao900914. I am using version `1.3.0`.

**Comment by bohan-amplitude** [comment]
Hi @mhernand40 Thank you for submitting feedback. I'm working on fix the issue. I'll release a new version and give you update.

**Comment by liuyang1520** [comment]
Thank you for supporting Amplitude! This issue should be fixed in https://github.com/amplitude/Amplitude-Kotlin/pull/103 and released in 1.6.1.
